### PR TITLE
MM-47192 : Migrate "utils/policy_roles_adapter.js" to typescript

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
Migrate utils/policy_roles_adapter.js to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21167
JIRA: https://mattermost.atlassian.net/browse/MM-47192

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: utils/policy_roles_adapter.ts [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
 // See LICENSE.txt for license information.
 
+import {Role} from '@mattermost/types/roles';
 import {Permissions} from 'mattermost-redux/constants/index';
 
 const MAPPING = {
     enableTeamCreation: {
-        true: [{roleName: 'system_user', permission: Permissions.CREATE_TEAM, shouldHave: true}],
-        false: [{roleName: 'system_user', permission: Permissions.CREATE_TEAM, shouldHave: false}],
+        True: [{roleName: 'system_user', permission: Permissions.CREATE_TEAM, shouldHave: true}],
```
Comment (M-ZubairAhmed) [inline utils/policy_roles_adapter.ts:9]: why did we change from 'true' to 'True'

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: utils/policy_roles_adapter.ts (hunk 1 above)
Comment (AbhinavVihan) [reply utils/policy_roles_adapter.ts:9]: Here true is creating conflicts, as it is a string as  well as Boolean.
So it's creating type mismatch errors in some places

Comment (AbhinavVihan) [reply utils/policy_roles_adapter.ts:9]: Here true is creating conflicts, as it is a string as  well as Boolean.
So it's creating type mismatch errors in some places

Comment (M-ZubairAhmed) [reply utils/policy_roles_adapter.ts:9]: ahh but its changing the behavior.

Comment (AbhinavVihan) [reply utils/policy_roles_adapter.ts:9]: another option was to add single quotes around true and disable lint warnings for the line, and give the type **'true' | 'false'** wherever required.

Comment (AshishDhama) [reply utils/policy_roles_adapter.ts:9]: @AbhinavVihan you can do ['true']

**Review by AshishDhama — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @AbhinavVihan 
Tested, looks good to merge.
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/248198-cb5ab2a-onprem-ent-webapp-pr-11246-mm-ee-test:cb5ab2a/mochawesome.html)- no PR related failures.

**Pull request merged**